docs/serverkeyformat.md

```diff
@@ -0,0 +1,29 @@
+# Server Key Format
+
+Dendrite stores the server signing key in the PEM format with the following structure.
+
+```
+-----BEGIN MATRIX PRIVATE KEY-----
+Key-ID: ed25519:<Key Handle>
+
+<Base64 Encoded Key Data>
+-----END MATRIX PRIVATE KEY-----
+```
+
+## Converting Synapse Keys
+
+If you have signing keys from a previous synapse server, you should ideally configure them as `old_private_keys` in your Dendrite config file. Synapse stores signing keys in the following format.
+
+```
+ed25519 <Key Handle> <Base64 Encoded Key Data>
+```
+
+To convert this key to Dendrite's PEM format, use the following template. **It is important to include the equals sign, as the key data needs to be padded to 32 bytes.**
```

Comment:
Is this true in all cases, or is it dependent on the key?

Reply:
From experimental testing, Dendrite complains about an early EOF if the key part of the PEM file is less than 32 bytes, probably because the PEM parser expects 32 bytes. Any key less than this length needs to be padded.

The length of the decoded base64 string generated by synapse seems to be 30 bytes, so keys generated by Synapse definetly have to be padded.